packages/sync/src/conflicts/conflictLink.ts

```diff
@@ -0,0 +1,51 @@
+import { ApolloLink } from "apollo-link";
+import { onError } from "apollo-link-error";
+import { GraphQLError } from "graphql";
+import { DataSyncConfig } from "../config/DataSyncConfig";
+import { ConflictResolutionData, strategies } from "./strategies";
+
+export class ConflictLink {
```

Comment:
[Optional] Any reason why this cannot be link itself (extend Apollo Link)?

Reply:
@wtrocki I spent yesterday evening trying to get this work as a class. I implemented it using observables under the request method but for some reason I could not get it to work as expected. When I tried to do it manually I could not trigger the errors to run the logic we need and when I tried to wrap onError in the request I lost access to graphQLErrors which we need to execute the same logic. I am reverting this to a function for now so we can merge and possibly come back to it.